rotkehlchen/exchanges/binance.py

```diff
@@ -530,6 +534,74 @@ def _query_spot_balances(
 
         return balances
 
+    def _query_funding_balances(
+            self,
+            balances: defaultdict[AssetWithOracles, Balance],
+    ) -> defaultdict[AssetWithOracles, Balance]:
+        """Query the balances of funding wallet in binance.
+        Docs: https://binance-docs.github.io/apidocs/spot/en/#funding-wallet-user_data
+
+        May Raise RemoteError due to api_query or invalid response"""
+        try:
+            funding_balances = self.api_query_list('sapi', 'asset/get-funding-asset', request_type='post')  # noqa: E501
+        except (RemoteError, BinancePermissionError) as e:
+            log.warning(
+                f'Failed to query {self.name} funding wallet balances.'
+                f'Skipping query. Response details: {e!s}',
+            )
+            return balances
+
+        if len(funding_balances) == 0:
+            return balances
+
+        for entry in funding_balances:
+            try:
+                asset_symbol = entry['asset']
+                free = deserialize_asset_amount(entry['free'])
+                locked = deserialize_asset_amount(entry['locked'])
+            except KeyError as e:
+                raise RemoteError(f'Binance funding balance asset entry did not contain key {e!s}') from e  # noqa: E501
+            except DeserializationError as e:
+                raise RemoteError('Failed to deserialize an amount from binance funding balance asset entry') from e  # noqa: E501
+
+            if (amount := free + locked) == ZERO:
+                continue
+
+            try:
+                asset = asset_from_binance(asset_symbol)
```

Comment:
This is entire try/except is done elsehwere in the code in this file. Please abstract it